src/Build/Evaluation/Expander.cs

```diff
@@ -2791,7 +2789,10 @@ internal static ItemTransformFunction GetItemTransformFunction(IElementLocation
 
                         try
                         {
-                            metadataValue = item.Value.GetMetadataValueEscaped(metadataName);
+                            if (item.Value.HasMetadata(metadataName))
```

Comment:
It might be nice to introduce a new "TryGetMetadataValueEscaped" function so you aren't effectively doing this twice.

Reply:
Yes, we can introduce such a new function. I have done that.  I didn't add this before since I think it would introduce more code changes.